src/pages/studyView/charts/scatterPlot/StudyViewDensityScatterPlot.tsx

```diff
@@ -0,0 +1,368 @@
+    @observable tooltipModel:any|null = null;
+    @observable pointHovered:boolean = false;
+    @observable mouseIsDown:boolean = false;
+    @observable shiftPressed:boolean = false;
+    public mouseEvents:any = makeMouseEvents(this);
+
+    @observable.ref private container:HTMLDivElement;
+    private svg:SVGElement|null;
+
+    @autobind
+    private containerRef(container:HTMLDivElement) {
+        this.container = container;
+    }
+
+    @autobind
+    private svgRef(svg:SVGElement|null) {
+        this.svg = svg;
+        if (this.props.svgRef) {
+            this.props.svgRef(this.svg);
+        }
+    }
+
+    @autobind
+    private onKeyDown(e: JQueryKeyEventObject) {
+        if (e.which === 16) {
+            this.shiftPressed = true;
+        }
+    }
+
+    @autobind
+    private onKeyUp(e: JQueryKeyEventObject) {
+        if (e.which === 16) {
+            this.shiftPressed = false;
+        }
+    }
+
+    componentDidMount() {
+        // Make it so that if you hold down shift, you can select more than one region at once
+        $(document).on("keydown",this.onKeyDown);
+        $(document).on("keyup", this.onKeyUp);
+    }
+
+    componentWillUnmount() {
+        $(document).off("keydown",this.onKeyDown);
+        $(document).off("keyup", this.onKeyUp);
+    }
+
+    public toSVGDOMNode(): Element {
+        return this.svg!;
+    }
+
+    private get title() {
+        if (this.props.title) {
+            return (
+                <VictoryLabel
+                    style={{
+                        fontWeight:"bold",
+                        textAnchor: "middle"
+                    }}
+                    x={this.props.width/2}
+                    y="1.2em"
+                    text={this.props.title}
+                />
+            );
+        } else {
+            return null;
+        }
+    }
+
+    @computed get dataDomain() {
+        // data extremes
+        const max = {x:Number.NEGATIVE_INFINITY, y:Number.NEGATIVE_INFINITY};
+        const min = {x:Number.POSITIVE_INFINITY, y:Number.POSITIVE_INFINITY};
+        for (const d of this.props.data) {
+            max.x = Math.max(d.x, max.x);
+            max.y = Math.max(d.y, max.y);
+            min.x = Math.min(d.x, min.x);
+            min.y = Math.min(d.y, min.y);
+        }
+        return {
+            //x: [min.x, max.x],
+            //y: [min.y, max.y]
+            x: [0, 1],
+            y:[0, max.y]
+        };
+    }
+
+    @computed get pixelSpaceToDataSpace() {
+        const dataDomain = this.dataDomain;
+        const xRange = dataDomain.x[1] - dataDomain.x[0];
+        const minX = dataDomain.x[0];
+        const yRange = dataDomain.y[1] - dataDomain.y[0];
+        const minY = dataDomain.y[0];
+        return {
+            x: (val:number)=>(val/this.props.width)*xRange + minX,
+            y: (val:number)=>(val/this.props.height)*yRange + minY
+        };
+    }
+
+    @computed get dataSpaceToPixelSpace() {
```

Comment:
The downsampling method should be moved to util function and add tests cover it.